step-templates/azure-web-app-restart.json

```diff
@@ -0,0 +1,44 @@
+{
+  "Id": "1f40e418-17bf-4b3e-bbe7-c6d41cbded93",
+  "Name": "Azure Web App - Restart",
+  "Description": "Restarts an azure web app.\n<hr />\n\n*<p>Note This template is designed to run against an azure web app octopus target </p>*\n*<p>Depends on Azure CLI and powershell to be installed on the running machine</p>*",
+  "ActionType": "Octopus.AzurePowerShell",
+  "Version": 1,
+  "CommunityActionTemplateId": null,
+  "Packages": [],
+  "Properties": {
+    "Octopus.Action.Script.ScriptSource": "Inline",
+    "Octopus.Action.Script.Syntax": "PowerShell",
+    "Octopus.Action.Azure.AccountId": "#{azWebApp.AzureAcct}",
+    "Octopus.Action.Script.ScriptBody": "try {az --version}\ncatch\n{\n    throw \"az CLI not installed\"\n}\n\n$webApp = $OctopusParameters[\"Octopus.Action.Azure.WebAppName\"]\n$resourceGroup = $OctopusParameters[\"Octopus.Action.Azure.ResourceGroupName\"]\n$startIfStopped = $OctopusParameters[\"azWebApp.StartIfStopped\"]\n\nWrite-Host \"Checking webapp $webApp status in resource group $resourceGroup\"\n\n$appState = az webapp list --resource-group chrisplayground --query \"[?name=='$webApp'].{state: state, hostName: defaultHostName}\" | ConvertFrom-Json\nif($appState.state -eq \"stopped\")\n{\n    if($startIfStopped) -eq 'true')\n    {\n        Write-Host \"Webapp is not running. Starting...\" -NoNewline\n    \taz webapp start --name $webApp --resource-group $resourceGroup\n        Write-Host \"Done\"\n    }\n    else\n    {\n      Throw \"Webapp is not running.\"\n    }\n}\n\nWrite-Host \"Webapp running, restarting\"\n\nelse\n{\n\tWrite-Host \"Restarting $webApp in resource group $resourceGroup\"\n\taz webapp restart --name $webApp --resource-group $resourceGroup\n}\n\nStart-Sleep -s 5\n\n$appState = az webapp list --resource-group chrisplayground --query \"[?name=='$webApp'].{state: state, hostName: defaultHostName}\" | ConvertFrom-Json\n\nif($appState.state -ne \"running\")\n{\n\tThrow \"Webapp failed to start.  Check the app's activity/error log\"\n}\n\nwrite-host \"Webapp $webApp running. Check at: $($appState.hostName)\"\n"
```

Comment:
Should the resource group be pointing to a hardcoded value here? 

`--resource-group chrisplayground`

Reply:
@harrisonmeister you are absolutly right i didn't scrub that out.  I'll go through these again and make sure the variables are being used.

Reply:
updated in commit bc42c4b